python_on_whales/components/image/cli_wrapper.py

```diff
@@ -503,49 +506,133 @@ def pull(
         self,
         x: Union[str, Iterable[str]],
         quiet: bool = False,
+        stream_logs: bool = False,
         platform: Optional[str] = None,
-    ) -> Union[Image, List[Image]]:
+    ) -> Union[Image, List[Image], Iterable[Tuple[str, bytes]]]:
         """Pull one or more docker image(s)
 
         Alias: `docker.pull(...)`
 
         Parameters:
-            x: The image name(s) . Can be a string or a list of strings. In case of
-                list, multithreading is used to pull the images.
-                The progress bars might look strange as multiple
-                processes are drawing on the terminal at the same time.
+            x: Image name(s), can be a string or a list of strings.
+                In case a list is passed, multithreading is used to pull
+                the images.
             quiet: If you don't want to see the progress bars.
+            stream_logs: If `False` this function returns the pulled image(s).
+                If `True`, this function returns an `Iterable` of `Tuple[str, bytes]`
+                where the first element corresponds to the image name that a pull is being done for.
+                The second element is the log statement related to pull progress, as `bytes`.
+                You'll need to call `.decode()` if you want the logs as `str`.
+                See [the streaming guide](https://gabrieldemarmiesse.github.io/python-on-whales/user_guide/docker_run/#stream-the-output)
+                if you are not familiar with the streaming of logs in Python-on-whales.
             platform: If you want to enforce a platform.
-
-        Returns:
-            The Docker image loaded (`python_on_whales.Image` object).
-            If a list was passed as input, then a `List[python_on_whales.Image]` will
-            be returned.
         """
+        if quiet and stream_logs:
+            raise ValueError(
+                "It's not possible to have stream_logs=True and quiet=True at the same time. "
+                "Only one can be activated at a time."
+            )
+
         if isinstance(x, str):
-            return self._pull_single_tag(x, quiet=quiet, platform=platform)
-        x = list(x)
-        if not x:
-            return []
-        if len(x) == 1:
-            return [self._pull_single_tag(x[0], quiet=quiet, platform=platform)]
+            result = self._pull_single_tag(
+                image_name=x,
+                quiet=quiet,
+                stream_logs=stream_logs,
+                platform=platform,
+            )
+            return result if stream_logs else Image(self.client_config, x)
+
+        images = list(dict.fromkeys(x))
```

Comment:
I'm not exactly sure why this is needed. Isn't 
```suggestion
        images = list(x)
```
fine too?

Reply:
Thanks for the explanation. Indeed that makes sense. I'll add it back, my bad